From 0d3552ba0c135830ebe2d6b0f410370b26491ac2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 23 Mar 2020 10:41:37 +0100
Subject: [PATCH 0590/1013] evl/proxy: do not discard undrained output upon
 disposal

We want the pending output to be drained upon disposal, so that write
operations we have already accepted do complete properly.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/proxy.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/kernel/evl/proxy.c b/kernel/evl/proxy.c
index 85eee78270eb..b1597226526e 100644
--- a/kernel/evl/proxy.c
+++ b/kernel/evl/proxy.c
@@ -387,7 +387,6 @@ static void proxy_factory_dispose(struct evl_element *e)
 	out = &proxy->output;
 	if (out->wq) {
 		irq_work_sync(&out->irq_work);
-		cancel_work_sync(&out->work);
 		destroy_workqueue(out->wq);
 	}
 	fput(proxy->filp);
-- 
2.30.2

